gax-httpjson/src/main/java/com/google/api/gax/httpjson/InstantiatingHttpJsonChannelProvider.java

```diff
@@ -168,12 +187,21 @@ private TransportChannel createChannel() throws IOException {
       headerEnhancers.add(HttpJsonHeaderEnhancers.create(header.getKey(), header.getValue()));
     }
 
+    HttpTransport httpTransportToUse = httpTransport;
+    if (httpTransportToUse == null) {
+      try {
+        httpTransportToUse = createHttpTransport();
+      } catch (GeneralSecurityException e) {
+        throw new IOException(e);
```

Comment:
again, this is in a private method so no need to wrap it yet. Changing the exception type is icky, so try to do it as late as possible.

Reply:
Done. Moved the wrapping to its caller `getTransportChannel`. This function overrides the one defined in `TransportChannelProvider` interface, so `getTransportChannel` is the last place to wrap the exception.